simulariumio/smoldyn/smoldyn_data.py

```diff
@@ -71,3 +79,70 @@ def __init__(
             spatial_units if spatial_units is not None else UnitData("m")
         )
         self.plots = plots if plots is not None else []
+
+    @classmethod
+    def from_dict(
+        cls, buffer_data: Dict[str, Any]
+    ):
+        """
+        Create SmoldynData from a simularium JSON dict containing buffers
+
+        Parameters
+        ----------
+        buffer_data: Dict[str, Any]
+            JSON dict containing values key value pairs representing the
+            data to be turned into a SmoldynData object
+        """
+        if (
+            "fileContents" not in buffer_data
+            or ("fileContents" not in buffer_data["fileContents"]
+                and "filePath" not in buffer_data["fileContents"])
+        ):
+            raise DataError(
+                "File contents or file path must be provided "
+                "to create a SmoldynData object"
+            )
+        display_data = None
+        if "displayData" in buffer_data:
+            display_data = unpack_display_data(buffer_data["displayData"])
+
+        spatial_units = None
+        if "spatialUnits" in buffer_data:
+            # spatial units defaults to meter in the UI
+            spatial_units = UnitData.from_dict(
+                buffer_data["spatialUnits"],
+                default_name="meter",
+                default_mag=1.0
+            )
+
+        time_units = None
+        if "timeUnits" in buffer_data:
+            # time units default to seconds on UI
+            time_units = UnitData.from_dict(
+                buffer_data["timeUnits"],
+                default_name="second",
+                default_mag=1.0
+            )
+
+        return cls(
+            meta_data=MetaData.from_dict(buffer_data.get("metaData")),
+            smoldyn_file=InputFileData(
+                file_contents=buffer_data["fileContents"].get("fileContents"),
```

Comment:
should these two have defaults?

Reply:
I currently have it set up so `InputFileData.from_dict()` will throw an error if `fileContents` is null, so I think I'd rather leave it without a default. `MetaData.from_dict()` will return a default `MetaData` object if called with null, so I think that's fine too